v0/destinations/woopra/transform.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-param-reassign */
+const { set } = require("lodash");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  getErrorRespEvents,
+  getIntegrationsObj,
+  getSuccessRespEvents,
+  defaultGetRequestConfig
+} = require("../../util");
+
+const { EventType } = require("../../../constants");
+const { BASE_URL, mappingConfig, ConfigCategories } = require("./config");
+const { refinePayload } = require("./utils");
+
+const responseBuilder = async (payload, endpoint, method, projectName) => {
+  if (!payload) {
+    throw new CustomError("[ WOOPRA ]:: Parameters could not be found", 400);
+  }
+  set(payload, "Project", projectName);
+  const response = defaultRequestConfig();
+  response.params = removeUndefinedAndNullValues(payload);
+  response.endpoint = endpoint;
+  response.method = method;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/identify`;
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  const prefix = ["cv"];
+  const method = defaultGetRequestConfig.requestMethod;
+  refinePayload(message, payload, prefix);
+  const extractedProjectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
+
+  return responseBuilder(payload, endpoint, method, extractedProjectName);
+};
+const trackResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/ce`;
+  if (!message.event) {
+    throw new CustomError("[ WOOPRA ]:: Event Name can not be empty.", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.TRACK.name]
+  );
+  const prefix = ["ce", "cv"];
```

Comment:
We do not need to add both ce and cv here. Just call the identifyResponseBuilder once from here. The user fields will be available. Do not call the responseBuilder function from inside this function. Keep it outside for all. and return the actual object that you have created